From 51a15108dd5433ac93145d1acd72e383fd7d8033 Mon Sep 17 00:00:00 2001
From: tgeisler <tgeisler@90b65887-3827-0410-9a23-83215b262276>
Date: Thu, 9 Jun 2011 14:16:26 +0000
Subject: [PATCH 0118/1076] massaiCorba V1.0.6.16 modification: changed orb
 connection handler behaviour to RW Fix for MANTIS:0019016: application hung,
 platform did not react (Emirates)

git-svn-id: svn://localhost/SelfServiceCommon/trunk@147 90b65887-3827-0410-9a23-83215b262276
---
 .../cpp/MassaiCorba/inc/massaiCORBA_def.h     | 15 ++++++++--
 .../Massai/cpp/MassaiCorba/src/corbaorb.cpp   | 30 ++++++++++++++++++-
 .../Massai/cpp/MassaiCorba/src/corbaorbEx.cpp | 12 +++++++-
 3 files changed, 52 insertions(+), 5 deletions(-)

diff --git a/SelfServiceCommon/Massai/cpp/MassaiCorba/inc/massaiCORBA_def.h b/SelfServiceCommon/Massai/cpp/MassaiCorba/inc/massaiCORBA_def.h
index 9bba3a51..3a54d307 100644
--- a/SelfServiceCommon/Massai/cpp/MassaiCorba/inc/massaiCORBA_def.h
+++ b/SelfServiceCommon/Massai/cpp/MassaiCorba/inc/massaiCORBA_def.h
@@ -11,10 +11,19 @@
 
 /*! \page massaicorbalib Massai CORBA library - Release Notes
  *
- *  @version 1.0.6.15
+ *  @version 1.0.6.16
  *
  *  \section change1 Change history
  *
+ *  \date 09.06.2011 \li V1.0.6.16
+ *                   \li GiT, modification: changed orb connection handler behaviour to RW
+ *                            TAO description : RW selects a strategy that simply blocks in recv() when waiting for a response from the server
+ *                                              instead of waiting in the Reactor using the Leader/Followers pattern.
+ *                            This change should prevent nested upcalls.
+ *                       To change the behaviour back to the old strategy set the environment variable
+ *                       STANDARD_ORB_WAIT_STRATEGY
+ *                   \li QA-Tests: All platform modules including the connection to applications are affected, an thorough integration test is needed
+ *
  *  \date 21.04.2011 \li V1.0.6.15
  *                   \li SteT, enhancement, added trace to corbaorb
  *
@@ -86,8 +95,8 @@
  *
  */
 
-// #define FILE_VER_BIN     1,0,6,15
-// #define FILE_VER_STR     "FileVersion", "1.0.6.15\0"
+// #define FILE_VER_BIN     1,0,6,16
+// #define FILE_VER_STR     "FileVersion", "1.0.6.16\0"
 
 // #define FILEDESCRIPTION  "FileDescription"  ,"Massai CORBA library\0"
 // #define INTERNALNAME     "InternalName"     ,MassaiCORBA"\0"
diff --git a/SelfServiceCommon/Massai/cpp/MassaiCorba/src/corbaorb.cpp b/SelfServiceCommon/Massai/cpp/MassaiCorba/src/corbaorb.cpp
index a11f31df..11269406 100644
--- a/SelfServiceCommon/Massai/cpp/MassaiCorba/src/corbaorb.cpp
+++ b/SelfServiceCommon/Massai/cpp/MassaiCorba/src/corbaorb.cpp
@@ -460,7 +460,7 @@ void corbaOrb::run()
 corbaOrb::corbaOrb()
 {
 int    argc= 0;
-char  *argv[20];
+char  *argv[33];
 long   port = 0;
 char   szArgs1[256];
 
@@ -499,6 +499,15 @@ char   szArgs1[256];
        {
          enableTAO_Logging(argc, argv);
        }
+
+      if(!getenv("STANDARD_ORB_WAIT_STRATEGY"))
+      {
+          argv[argc++] = "-ORBSvcConfDirective";
+          argv[argc++] = "static Client_Strategy_Factory \"-ORBClientConnectionHandler rw\"";
+          argv[argc++] = "-ORBSvcConfDirective";
+          argv[argc++] = "static Resource_Factory \"-ORBFlushingStrategy blocking\"";
+      }
+
        if(port > 0)
        {
          sprintf(szArgs1,"iiop://:%d",port); // bind TAO-ORB to this port
@@ -632,10 +641,20 @@ char   szArgs1[256];
        argv[argc++] = "-ORBKeepalive";
        argv[argc++] = "1";
 #endif
+
        if(getenv("TAO_ENABLE_LOGGING"))
        {
          enableTAO_Logging(argc, argv);
        }
+
+      if(!getenv("STANDARD_ORB_WAIT_STRATEGY"))
+      {
+          argv[argc++] = "-ORBSvcConfDirective";
+          argv[argc++] = "static Client_Strategy_Factory \"-ORBClientConnectionHandler rw\"";
+          argv[argc++] = "-ORBSvcConfDirective";
+          argv[argc++] = "static Resource_Factory \"-ORBFlushingStrategy blocking\"";
+      }
+
        if(port <= 0)
        {
          if(getenv("ORBENDPOINT_PORT"))
@@ -776,6 +795,15 @@ char   szArgs1[256];
        {
          enableTAO_Logging(argc, argv);
        }
+
+      if(!getenv("STANDARD_ORB_WAIT_STRATEGY"))
+      {
+          argv[argc++] = "-ORBSvcConfDirective";
+          argv[argc++] = "static Client_Strategy_Factory \"-ORBClientConnectionHandler rw\"";
+          argv[argc++] = "-ORBSvcConfDirective";
+          argv[argc++] = "static Resource_Factory \"-ORBFlushingStrategy blocking\"";
+      }
+
        if(port <= 0)
        {
          if(getenv("ORBENDPOINT_PORT"))
diff --git a/SelfServiceCommon/Massai/cpp/MassaiCorba/src/corbaorbEx.cpp b/SelfServiceCommon/Massai/cpp/MassaiCorba/src/corbaorbEx.cpp
index 10bf3c28..d948ace5 100644
--- a/SelfServiceCommon/Massai/cpp/MassaiCorba/src/corbaorbEx.cpp
+++ b/SelfServiceCommon/Massai/cpp/MassaiCorba/src/corbaorbEx.cpp
@@ -215,7 +215,7 @@ long corbaOrbEx::createOrb()
 {
     long rc;
     int         argc;
-    char       *argv       [20];
+    char       *argv       [33];
     char        szArgs1    [256];
 
 
@@ -237,6 +237,16 @@ long corbaOrbEx::createOrb()
       {
         enableTAO_Logging(argc, argv);
       }
+
+      if(!getenv("STANDARD_ORB_WAIT_STRATEGY"))
+      {
+          argv[argc++] = "-ORBSvcConfDirective";
+          argv[argc++] = "static Client_Strategy_Factory \"-ORBClientConnectionHandler rw\"";
+          argv[argc++] = "-ORBSvcConfDirective";
+          argv[argc++] = "static Resource_Factory \"-ORBFlushingStrategy blocking\"";
+      }
+
+
       // provide different orb names if we have more than one orb in one process !!!
 
       sprintf(szOrbName,"ORB$%04x$%d",GetCurrentThreadId(),lOrbCnt);
-- 
2.41.0.windows.1

